### DISCUSSION TITLE
How to distinguish between right click and left click?

### DISCUSSION THREAD

**Discussion by francescvc** [AlmasB/FXGL#1099] [Q&A]
On entityBuilder() I can  setup an  onClick((ent)-> { ... })  but on the click code I can't make difference upon the button clicked, that is the left one or right one.  ¿How can I attach different events to both mousebuttons, or how can I distinguish wich one was clicked?

**Comment by AlmasB** [comment]
It sounds like there should be `onClick` and `onRightClick` or similar. Feel free to open an issue.

In the meantime, you can use `entity.getViewComponent().addEventHandler ( ... )` as you normally would with javafx

**Reply by chengenzhao** [reply]
People keep asking how to handle the mouse event with entity object
Would be nice to have all mouse event short cut method on entity object
similar to entity.onClick
like entity.onPressed, entity.onDragged, entity.onReleased... etc.

**Reply by chengenzhao** [reply]
Some people suggest to have a Mouse Event Component or Mouse Component which handles mouse events
use it like:
entity.with(new MouseEventComponent())

**Reply by AlmasB** [reply]
> People keep asking how to handle the mouse event with entity object Would be nice to have all mouse event short cut method on entity object similar to entity.onClick like entity.onPressed, entity.onDragged, entity.onReleased... etc.

EntityBuilder is DSL on top of Entity, so we can add it there. I'll turn this into an issue.
